src/overlays/actors/ovl_En_Sekihi/z_en_sekihi.c

```diff
@@ -32,20 +36,150 @@ const ActorInit En_Sekihi_InitVars = {
+
+    if ((params < SEKIHI_TYPE_0 || params >= SEKIHI_TYPE_MAX) || sOpaDLists[params] == NULL) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        if ((params == SEKIHI_TYPE_4) && (((gSaveContext.save.skullTokenCount & 0xFFFF)) >= 30)) {
+            gSaveContext.save.weekEventReg[13] |= 0x20;
+        }
+        objectIndex = Object_GetIndex(&play->objectCtx, sObjectIds[params]);
+        if (objectIndex >= 0) {
+            this->objectIndex = objectIndex;
+        }
+        this->actionFunc = func_80A44DE8;
+        this->opaDList = sOpaDLists[params];
+        this->xluDList = sXluDLists[params];
+        this->dyna.actor.textId = sTextIds[params];
+        this->dyna.actor.focus.pos.y = this->dyna.actor.world.pos.y + 60.0f;
+        Actor_SetScale(&this->dyna.actor, 0.1f);
+    }
+}
+
+void EnSekihi_Destroy(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80A44DE8(EnSekihi* this, PlayState* play) {
+    CollisionHeader* colHeader = NULL;
+    s32 params;
+    CollisionHeader* colHeaders[] = {
+        &gSekihilCol, &gSekihigCol, &gSekihinCol, &gSekihizCol, &gObjectZogCol,
+    };
+
+    params = ENSIKIHI_GET_TYPE(&this->dyna.actor);
+    if (Object_IsLoaded(&play->objectCtx, this->objectIndex)) {
+        this->dyna.actor.objBankIndex = this->objectIndex;
+        this->dyna.actor.draw = EnSekihi_Draw;
+        if (params == SEKIHI_TYPE_4) {
+            this->actionFunc = func_80A450B0;
+        } else {
+            this->actionFunc = EnSekihi_DoNothing;
+        }
+
+        Actor_SetObjectDependency(play, &this->dyna.actor);
+        DynaPolyActor_Init(&this->dyna, 0);
+        if (colHeaders[params] != NULL) {
+            CollisionHeader_GetVirtual(colHeaders[params], &colHeader);
+        }
+
+        this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+        if ((params == SEKIHI_TYPE_4) && (INV_CONTENT(ITEM_MASK_ZORA) != ITEM_MASK_ZORA)) {
+            Actor_MarkForDeath(&this->dyna.actor);
+        }
+    }
+}
+
+void func_80A44F40(EnSekihi* this, PlayState* play) {
+    switch (Message_GetState(&play->msgCtx)) {
+        case 4:
+            if ((Message_ShouldAdvance(play)) && (play->msgCtx.currentTextId == 0x1019)) {
+                switch (play->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        func_80151938(play, 0x101A);
+                        break;
+                    case 1:
+                        func_8019F208();
+                        func_80151938(play, 0x101B);
+                        break;
+                    case 2:
+                        func_8019F230();
+                        func_801477B4(play);
+                        this->actionFunc = func_80A450B0;
+                        break;
+                }
+            } else {
+                break;
+            }
```

Comment:
```suggestion
            }
```
Is the else actually needed?

Reply:
yes it is